### PR TITLE
[WNMGDS-1871] Migrate unit tests for Mask to RTL

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

https://jira.cms.gov/browse/WNMGDS-1871

## How to test

`yarn test:unit`

### If this is a change to code:

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[ticket] title`
- [x] Ran `yarn type-check && yarn lint && yarn test` with no errors
- [x] Updated unit test snapshots and loki reference images **if necessary** with `yarn loki` && `yarn update-snapshots`

### CODE REVIEW

File: packages/design-system/src/components/TextField/Mask.test.tsx [hunk 1]
```diff
@@ -1,153 +1,131 @@
...
     </Mask>
   );
+}
 
-  return {
-    props: customProps,
-    wrapper: deep ? mount(component) : shallow(component),
-  };
+function getInput() {
+  return screen.getByRole('textbox') as HTMLInputElement;
```
Comment (forrestbaer) [inline packages/design-system/src/components/TextField/Mask.test.tsx:19]: I gotta remember to cast these in the future I was trying to figure out how to do this earlier.

File: packages/design-system/src/components/TextField/Mask.test.tsx [hunk 2]
```diff
@@ -1,153 +1,131 @@
...
-    wrapper
-      .find('input')
-      .props()
-      .onBlur({ target: { value: '1234' }, persist: jest.fn() });
-
-    expect(onBlur.mock.calls.length).toBe(1);
+    renderMask({ mask: 'currency' }, { value: '123', onBlur });
+    userEvent.click(getInput());
+    userEvent.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
```
Comment (forrestbaer) [inline packages/design-system/src/components/TextField/Mask.test.tsx:57]: Gah this is so much more succint, love it.

File: packages/design-system/src/components/TextField/Mask.test.tsx [hunk 3]
```diff
@@ -1,153 +1,131 @@
...
+            />
+          </Mask>
+        );
+      }
+
+      render(<ControlledComponentParent />);
+      expect(getInput().value).toBe('1,000');
+
+      // Remove the comma
+      userEvent.type(getInput(), '{arrowleft}{arrowleft}{arrowleft}{backspace}');
```
Comment (forrestbaer) [inline packages/design-system/src/components/TextField/Mask.test.tsx:94]: Oh wow didn't know about this either.

**Review by forrestbaer — COMMENTED** [review]
Looks great! Just an unresolved merge conflict at the top of `/packages/design-system/src/components/TextField/TextField.test.tsx`

edit: hmm that was something strange in my branch.. Looks good.

**Review by forrestbaer — APPROVED** [review]

File: packages/design-system/src/components/TextField/Mask.test.tsx (hunk 3 above)
Comment (pwolfert) [reply packages/design-system/src/components/TextField/Mask.test.tsx:94]: It feels like a cheat code 😆 `^^vv<><>...`

**Pull request merged**